src/pcp/meminfo/pcp-meminfo.py

```diff
@@ -0,0 +1,211 @@
+                "WritebackTmp",
+                "CommitLimit",
+                "Committed_AS",
+                "VmallocTotal",
+                "VmallocUsed",
+                "VmallocChunk",
+                "HardwareCorrupted",
+                "AnonHugePages",
+                "ShmemHugePages",
+                "ShmemPmdMapped",
+                "CmaFree",
+                "HugePages_Total_NO_kb",
+                "HugePages_Free_NO_kb",
+                "HugePages_Rsvd_NO_kb",
+                "HugePages_Surp_NO_kb",
+                "Hugepagesize",
+                "DirectMap4k",
+                "DirectMap2M",
+                "DirectMap1G"]
+
+class MeminfoReport(pmcc.MetricGroupPrinter):
+
+    def getMetricName(self, idx):
+        metric_name = ""
+        units = ""
+        if METRICS_DESC[idx][-6:] == "_NO_kb":
+            metric_name = METRICS_DESC[idx][:-6]
+        else:
+            metric_name = METRICS_DESC[idx]
+            units = "kB"
+        return metric_name, units
+
+    def report(self, manager):
+        group = manager["meminfo"]
+        t_s = group.contextCache.pmLocaltime(int(group.timestamp))
+        time_string = time.strftime(MeminfoOptions.timefmt, t_s.struct_time())
+        print(time_string)
+
+        if MeminfoOptions.samples == 0 and MeminfoOptions.context is PM_CONTEXT_ARCHIVE:
+            MeminfoOptions.samples = sys.maxsize
+
+        idx = 0
+        for metric in METRICS:
+            try:
+                val = group[metric].netValues[0][2]
+            except IndexError:
+                metric_name, units = self.getMetricName(idx)
+                print(F"{metric_name:17} : NA")
+
+                idx += 1
+                continue
+
+            metric_name, units = self.getMetricName(idx)
+            print(F"{metric_name:17} : {val} {units}")
+
+            idx += 1
+        print()
+
+        MeminfoOptions.samples -= 1
+
+        if MeminfoOptions.samples <= 0:
+            sys.exit(0)
+
+class MeminfoOptions(pmapi.pmOptions):
+    context = None
+    timefmt = "%H:%M:%S"
+    samples = 0
+
+    def __init__(self):
+        pmapi.pmOptions.__init__(self, "a:s:S:T:z:P:A:R:t:T:x:")
+        self.pmSetLongOption("archive", 1, "a", "FILENAME","Fetch /proc/meminfo for a specified archive file")
+        self.pmSetLongOption("samples", 1, "s", "count","Get the meminfo for specified number of samples count")
+        self.pmSetLongOption("start_time", 1, "S", "TIME","Filter the samples from the archive from the given time")
+        self.pmSetLongOption("end_time", 1, "T", "TIME","Filter the samples from the archive till the given time")
+        self.pmSetOptionCallback(self.options)
+        self.pmSetOverrideCallback(self.override)
+        self.pmSetLongOptionHelp()
+
+    def override(self, opt):
+        if opt == 's':
+            return 1
+        return 0
+
+    def options(self,opt,optarg,index):
+        if opt == 's':
+            MeminfoOptions.samples = int(optarg)
```

Comment:
Ah I see why you handle 's' differently now - you don't need this either, use opts.pmGetOptionSamples() at around line 140 (you could pass this value into a MeminfoReport __init__ function, called at line ~201).

Reply:
Hi scott, 

I tried getting number of samples using the opts.pmGetOptionSamples(), but it was not able to retrieve the number successfully, instead it's showing "None" no matter how many number of samples are passed. So, as a workaround to that, I implemented it in this manner. Anyways, let me take a look at it again if I can modify this using opts.pmGetOptionSamples(). 

Thanks!

Reply:
@mohith-kumar-thummaluru make sure you access it after option-processing has been completed - because of code abstractions its not always obvious where that happens though.  One safe place would be at the point the MeminfoReport object is created as this happens relatively late in the life of the program.